spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -1,2 +1,217 @@
+
+Assume an Order entity has an Address with a Zipcode. In that case, the access is ```order.address.zipCode```.
+
+The resolution algorithm starts by interpreting the whole part (AddressZipCode) as the property and checks the domain class for a property with that name (uncapitalized). If the algorithm succeeds, it uses that property.
+
+[source,java]
+----
+@Repository
+public interface OrderRepository extends CrudRepository<Order, Long> {
+
+  Stream<Order> findByAddressZipCode(ZipCode zipCode);
+
+}
+----
+
+
+Although this should work for most cases, to resolve this ambiguity, you can use ```_``` inside your method name to manually define traversal points.
+
+[source,java]
+----
+@Repository
+public interface OrderRepository extends CrudRepository<Order, Long> {
+
+  Stream<Order> findByAddress_ZipCode(ZipCode zipCode);
+
+}
+----
+
+WARNING: Define as a priority following standard Java naming conventions, camel case,  using underscore as the last instance resource.
+
+===== Query methods Keywords
+
+The following table lists the subject keywords generally supported by Jakarta Data.
+
+|===
+|Keyword |Description
+
+|findBy
+|General query method returning the repository type.
+
+|deleteBy
+|Delete query method returning either no result (void) or the delete count.
+
+|countBy
+|Count projection returning a numeric result.
+
+|existsBy
+|Exists projection, returning typically a ```boolean``` result.
+|===
+
+Jakarta Data often support the following list of predicate keywords. It might not work because some keywords listed here might not be supported in a particular store.
+
+|===
+|Keyword |Description | Method signature Sample
+
+|And
+|The ```and``` operator.
+|findByNameAndYear
+
+|Or
+|The ```or``` operator.
+|findByNameOrYear
+
+|Between
+|Find results where the property is between the given values
+|findByDateBetween
+
+|LessThan
+|Find results where the property is less than the given value
+|findByAgeLessThan
+
+|GreaterThan
+|Find results where the property is greater than the given value
+|findByAgeGreaterThan
+
+|LessThanEqual
+|Find results where the property is less than or equal to the given value
+|findByAgeLessThanEqual
+
+|GreaterThanEqual
+|Find results where the property is greater than or equal to the given value
+|findByAgeGreaterThanEqual
+
+|Like
+|Finds string values "like" the given expression
+|findByTitleLike
+
+|In
+|Find results where the property is that are contained within the given list
+|findByIdIn
+
+|OrderBy
+|Specify a static sorting order followed by the property path and direction on ascending.
+|findByNameOrderByAge
+
+|OrderBy____Desc
+|Specify a static sorting order followed by the property path and direction on descending.
+|findByNameOrderByAgeDesc
+
+|OrderBy____Asc
+|Specify a static sorting order followed by the property path and direction on ascending.
+|findByNameOrderByAgeAsc
```

Comment:
Need to clarify if ordering on multiple attributes is done as
```
findBy...OrderByAgeDescLastNameAscFirstNameAsc
```
or
```
findBy...OrderByAgeDescOrderByLastNameAscOrderByFirstNameAsc
```

Note that in the first case, it might be necessary to require Asc or Desc to delimit multiple. In the second case, the method name gets a lot longer.  I'm also thinking to write an issue at some point to see what others think about defining OrderBy annotatively, which could help avoid lengthy method names.

Reply:
Thanks, I'updated